test/src/tests-vault.ts

```diff
@@ -0,0 +1,271 @@
+import { Batch } from '@mparticle/event-models';
+import { expect } from 'chai';
+import Vault from '../../src/vault';
+
+describe.only('Vault', () => {
+    afterEach(() => {
+        window.localStorage.clear();
+    });
+
+    describe('#storeBatches', () => {
+        it('should store an array of batches as a hashmap', () => {
+            const batch1 = {
+                foo: 'bar',
+                source_request_id: 'item-123',
+            };
+            const batch2 = {
+                foo: 'bar',
+                source_request_id: 'item-456',
+            };
+
+            const batches = ([batch1, batch2] as unknown) as Batch[];
+
+            const vault = new Vault('test-key-store-batches');
+
+            const expectedContents = { 'item-123': batch1, 'item-456': batch2 };
+
+            vault.storeBatches(batches);
+
+            expect(vault.contents).to.eql(expectedContents);
+        });
+
+        it('should update an existing batches with new events', () => {
+            const batch1 = {
+                foo: 'bar',
+                source_request_id: 'item-123',
+                events: [
+                    { eventName: 'test-event-1' },
+                    { eventName: 'test-event-2' },
+                ],
+            };
+
+            const batch2 = {
+                foo: 'bar',
+                source_request_id: 'item-456',
+                events: [{ eventName: 'test-event-3' }],
+            };
+
+            const batches = ([batch1, batch2] as unknown) as Batch[];
+
+            const vault = new Vault('test-key-store-batches');
+
+            const expectedContents = {
+                'item-123': {
+                    foo: 'bar',
+                    source_request_id: 'item-123',
+                    events: [
+                        { eventName: 'test-event-1' },
+                        { eventName: 'test-event-2' },
+                        { eventName: 'test-event-4' },
+                    ],
+                },
+                'item-456': {
+                    foo: 'bar',
+                    source_request_id: 'item-456',
+                    events: [
+                        { eventName: 'test-event-3' },
+                        { eventName: 'test-event-5' },
+                    ],
+                },
+            };
```

Comment:
Similar to above, for readability, this block of code should go right before the expect statement on line 79.